File: python/prograimg_python_basic/Exercises/conditional_statements/conditional_statements_advanced_-_exercise/operations_between_numbers.py
n1 = int(input())
n2 = int(input())
operator = input()

result = ""
sum = 0

if n2 == 0:
    print(f"Cannot divide {n1} by zero")

# •	Ако операцията е модулно деление
elif operator == "%":
    sum = n1 % n2

# •	Ако операцията е събиране, изваждане или умножение:
elif operator == "+":
    sum = n1 + n2
    if sum % 2 == 0:
        result = "even"
    else:
        result = "odd"

elif operator == "-":
    sum = n1 - n2
    if sum % 2 == 0:
        result = "even"
    else:
        result = "odd"

elif operator == "*":
    sum = n1 * n2
    if sum % 2 == 0:
        result = "even"
    else:
        result = "odd"

# •	Ако операцията е деление:
elif operator == "/":
    sum = "{:.2f}".format(n1 / n2)

if result == "" and n2 != 0:
    print(f"{n1} {operator} {n2} = {sum}")

elif result == "even" or result == "odd" and n2 != 0:
    print(f"{n1} {operator} {n2} = {sum} - {result}")

File: JS/JavaScript/JS-Basics/ConditionalStatementsAdvancedExercise/OperationsBetweenNumbers.js
function calculator(inputs) {

    let numberOne = Number(inputs[0])
    let numberTwo = Number(inputs[1])
    let operator = String(inputs[2])

    let result;
    let evenOrOdd;

    if (operator == "+") {
        sum = numberOne + numberTwo;
        if (sum % 2 == 0) {
            evenOrOdd = "even";
        }
        else {
            evenOrOdd = 'odd';
        }
        result = `${numberOne} ${operator} ${numberTwo} = ${sum} - ${evenOrOdd}`;
    }
    else if (operator == "-") {
        sum = numberOne - numberTwo;
        if (sum % 2 == 0) {
            evenOrOdd = "even";
        }
        else {
            evenOrOdd = 'odd';
        }
        result = `${numberOne} ${operator} ${numberTwo} = ${sum} - ${evenOrOdd}`;
    }
    else if (operator == "*") {
        sum = numberOne * numberTwo;
        if (sum % 2 == 0) {
            evenOrOdd = "even";
        }
        else {
            evenOrOdd = 'odd';
        }
        result = `${numberOne} ${operator} ${numberTwo} = ${sum} - ${evenOrOdd}`;
    }
    else if (operator == "/") {
        if (numberOne == 0 ) {
            result = `Cannot divide ${numberTwo} by zero`;
        }
        else if (numberTwo == 0) {
            result = `Cannot divide ${numberOne} by zero`;
        }
        else {
            sum = (numberOne / numberTwo).toFixed(2);
            result = `${numberOne} ${operator} ${numberTwo} = ${sum}`;
        };
    }
    else if (operator == "%") {
        if (numberOne == 0 ) {
            result = `Cannot divide ${numberTwo} by zero`;
        }
        else if (numberTwo == 0) {
            result = `Cannot divide ${numberOne} by zero`;
        }
        else {
            sum = numberOne % numberTwo;
            result = `${numberOne} ${operator} ${numberTwo} = ${sum}`;
        };
    };

    console.log(result);
}

calculator(["10","12","+"])
calculator(["123","12","/"])
calculator(["112","0","/"])
calculator(["10","0","/"])
calculator(["10","1","-"])


